src/molecules/formfields/util/BaseDateField/index.js

```diff
@@ -38,6 +38,19 @@ class BaseDateField extends React.Component {
     }
   }
 
+  createDate = (date) => {
+    const parsedDate = date.split('-');
+    if (parsedDate.length === 3) {
+      const year = parsedDate[0];
+      const month = parsedDate[1];
+      // Split day with timestamp if it is in ISO-8601 format
+      const day = parsedDate[2].split('T')[0];
+      return new Date(year, month - 1, day);
+    } else {
+      return new Date(date);
```

Comment:
If for whatever reason we can't parse this date, i'm just passing it directly into `new Date()` and let it handle it. Doubt it'd ever happen though.